### PR TITLE
refactor: split parse component and entity

### PR DESCRIPTION
**Pull request by gz65555**
Fix https://github.com/ant-galaxy/editor/issues/1154，clearly specify the process of the parsing:
```typescript
start() {
    this._parseEntities()
      .then(this._organizeEntities)
      .then(this._parseComponents)
      .then(this._clearAndResolveScene)
      .then(this._resolve)
      .catch(this._reject);
 }
```

### CODE REVIEW

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+    texture: null;
+    sky: null;
+  };
+  ambient: {
+    ambientLight: null;
+    diffuseSolidColor: { r: 0.212; g: 0.227; b: 0.259; a: 1 };
+    diffuseIntensity: 1;
+    specularIntensity: 1;
+  };
 }
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:19]: IPrefabFile may not need to contain Background and Ambient

File: packages/loader/src/resource-deserialize/resources/prefab/ReflectionParser.ts [hunk 2]
```diff
@@ -12,30 +12,10 @@ export class ReflectionParser {
...
-          component = entity.getComponent(Loader.getClass(key));
-        }
-        component = component || entity.addComponent(Loader.getClass(key));
-        const promise = this.parsePropsAndMethods(component, componentConfig, engine);
-        promises.push(promise);
-      }
-      return Promise.all(promises).then(() => {
-        return entity;
-      });
+      if (position) entity.transform.setPosition(position.x, position.y, position.z);
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/prefab/ReflectionParser.ts:15]: use `entity.transform.position.copyFrom(position);` is better

File: packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts [hunk 3]
```diff
@@ -1,36 +1,113 @@
...
-    const entitiesMap: Record<string, Entity> = {};
-    const entitiesConfigMap: Record<string, IEntity> = {};
-    const promises: Promise<Entity>[] = [];
-    const entitiesConfig = sceneData.entities;
-    for (const entity of entitiesConfig) {
-      entitiesConfigMap[entity.id] = entity;
-      promises.push(ReflectionParser.parseEntity(entity, engine));
-    }
+    const context = new SceneParserContext(sceneData, scene);
+    const parser = new SceneParser(context);
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts:18]: `SceneParser` maybe should not create multi times,use static or just create once

File: packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts [hunk 4]
```diff
@@ -1,36 +1,113 @@
...
+
+  private _parseEntities(): Promise<Entity[]> {
+    const entitiesConfig = this.context.originalData.entities;
+    const entityConfigMap = this.context.entityConfigMap;
+    const entitiesMap = this.context.entityMap;
+    const rootIds = this.context.rootIds;
+    const engine = this._engine;
+    const promises = entitiesConfig.map((entityConfig) => {
+      entityConfigMap.set(entityConfig.id, entityConfig);
+      // 记录 root entity
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts:61]: Use english!

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts (hunk 1 above)
Comment (gz65555) [reply packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:19]: done

File: packages/loader/src/resource-deserialize/resources/prefab/ReflectionParser.ts (hunk 2 above)
Comment (gz65555) [reply packages/loader/src/resource-deserialize/resources/prefab/ReflectionParser.ts:15]: done

File: packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts (hunk 4 above)
Comment (gz65555) [reply packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts:61]: done

**Pull request merged**